ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/eventstream/collection/InMemoryEventStreamCollection.java

```diff
@@ -0,0 +1,52 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.domain.eventstream.collection;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.eventstream.repository.EventStreamRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.eventstream.valueobjects.EventStream;
+import jakarta.annotation.PostConstruct;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class InMemoryEventStreamCollection implements EventStreamCollection {
+	private final Map<String, EventStream> eventStreams = new HashMap<>();
+	private final EventStreamRepository repository;
+
+	public InMemoryEventStreamCollection(EventStreamRepository repository) {
+		this.repository = repository;
+	}
+
+	@Override
+	public List<EventStream> retrieveAllEventStreams() {
+		return List.copyOf(eventStreams.values());
+	}
```

Comment:
Nice! We should use this more (returning immutable objects) to avoid that other classes secretly change internals attributes.